### PR TITLE
[patch_subset] Add ability to selective transfer opentype layout features.

### LINKED ISSUE #87: Support OpenType Features (discretionary)
It seems while IFT currently supports transferring OpenType Variations, it doesn't support transferring OpenType Features (that aren't mandatory; I understand those are transferred as part of the data needed to support requested unicode characters. )

For example, say that I have a font file which supports small caps, and I want to start using them.....

https://w3c.github.io/IFT/Overview.html#PatchRequest has

ID | Field Name | Value Type
-- | -- | --
2 | codepoints_have | CompressedSet
3 | codepoints_needed | CompressedSet
4 | indices_have | CompressedSet
5 | indices_needed | CompressedSet
**6** | **axis_space_have** | **AxisSpace**
**7** | **axis_space_needed** | **AxisSpace**

I expect

ID | Field Name | Value Type
-- | -- | --
2 | codepoints_have | CompressedSet
3 | codepoints_needed | CompressedSet
4 | indices_have | CompressedSet
5 | indices_needed | CompressedSet
6 | axis_space_have | AxisSpace
7 | axis_space_needed | AxisSpace
**9** | **features_have** | **FeatureList**
**10** | **features_needed** | **FeatureList**

(Possibly this related practically to adding Font Collection support #28)

### PR DESCRIPTION
**Pull request by garretrieger**
Adds new fields to PatchRequest/Response to allow the client to selectively request opentype layout features:

- There's a 'default' set of features which are always sent. These were selected based on the "default on features" listed in http://tiro.com/John/Enabling_Typography_(OTL).pdf
- If the fields are not set this acts like requesting all features.
- I've added an appendix B which is a registry of features which includes whether they are default or not plus an ID for efficiently encoding the non-default features.
- Appendix B is generated from "feature-registry.csv" as part of the spec build.
- Updated rendering equivalence requirement to mention the feature and VF axis subset.
- Fixes #87

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/IFT/pull/95.html" title="Last updated on Jun 3, 2022, 5:45 PM UTC (d14f06d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/IFT/95/7184356...d14f06d.html" title="Last updated on Jun 3, 2022, 5:45 PM UTC (d14f06d)">Diff</a>

### CODE REVIEW

File: feature-registry.csv [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+cpct,Centered CJK Punctuation,0
+cpsp,Capital Spacing,0
+cswh,Contextual Swash,0
+curs,Cursive Positioning,1
+cv01-cv99,Character Variants,0
+c2pc,Petite Capitals From Capitals,0
+c2sc,Small Capitals From Capitals,0
+dist,Distances,1
+dlig,Discretionary Ligatures,0
+dnom,Denominators,0
```
Comment (khaledhosny) [inline feature-registry.csv:28]: HarfBuzz enabled `frac`, `numr`, and `dnom` automatically on numbers surrounding a fraction slash, so stripping these features by default will change rendering.

File: feature-registry.csv [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+rand,Randomize,1
+rclt,Required Contextual Alternates,1
+rkrf,Rakar Forms,1
+rlig,Required Ligatures,1
+rphf,Reph Forms,1
+rtbd,Right Bounds,0
+rtla,Right-to-left alternates,1
+rtlm,Right-to-left mirrored forms,1
+ruby,Ruby Notation Forms,0
+rvrn,Required Variation Alternates,#N/A
```
Comment (khaledhosny) [inline feature-registry.csv:96]: Default.

File: feature-registry.csv [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+salt,Stylistic Alternates,0
+sinf,Scientific Inferiors,0
+size,Optical size,#N/A
+smcp,Small Capitals,0
+smpl,Simplified Forms,0
+ss01-ss20,Stylistic Set 1,0
+ssty,Math script style alternates,#N/A
```
Comment (khaledhosny) [inline feature-registry.csv:103]: Default in math layout.

File: feature-registry.csv [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+cpsp,Capital Spacing,0
+cswh,Contextual Swash,0
+curs,Cursive Positioning,1
+cv01-cv99,Character Variants,0
+c2pc,Petite Capitals From Capitals,0
+c2sc,Small Capitals From Capitals,0
+dist,Distances,1
+dlig,Discretionary Ligatures,0
+dnom,Denominators,0
+dtls,Dotless Forms,#N/A
```
Comment (khaledhosny) [inline feature-registry.csv:29]: Default in math layout.

File: feature-registry.csv [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+expt,Expert Forms,0
+falt,Final Glyph on Line Alternates,0
+fin2,Terminal Forms #2,1
+fin3,Terminal Forms #3,1
+fina,Terminal Forms,1
+flac,Flattened accent forms,#N/A
```
Comment (khaledhosny) [inline feature-registry.csv:35]: Default in math layout.

File: Overview.bs [hunk 6]
```diff
@@ -805,6 +838,13 @@ transferred:
 *  Subset Axis Space: the most recent variations axis space that the subsetted font covers. Supplied by
     <a href="#PatchResponse"><code>PatchResponse.subset_axis_space</code></a>.
 
+Additionally, the client can optionally store:
+
+*  Original font feature list: a list of the
+    <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/featuretags">opentype layout feature tags</a>
+    that the original font has data for. This information can be used by the client to avoid sending
+    unnessecary requests for features which the original font does not contain.
```
Comment (svgeesus) [inline Overview.bs:846]: spelling :) unnecessary

File: feature-registry.csv [hunk 7]
```diff
@@ -0,0 +1,126 @@
+# New features should be added to the bottom to not disrupt existing numbering.
```
Comment (svgeesus) [inline feature-registry.csv:1]: Where does the "on or off by default" data come from?

Comment (garretrieger) [reply feature-registry.csv:18]: This is a combination of John's list of default on features, the set of default features from harfbuzz subset, plus the features that Kahled mentioned being needed for math layout. I've added a comment to this file that documents that source of the data.

**Review by vlevantovsky — APPROVED** [review]
Looks good with previously submitted comments already addressed.

**Pull request merged**